### DISCUSSION TITLE
export full database

### DISCUSSION THREAD

**Discussion by farfromrefug** [osm-search/Nominatim#2692] [Q&A]
Hi is there a way to export the database as JSON?

I am looking at using nominatim to "prepare" geocoding data to be use  in an offline geocoding library (for mobile devices).
nominatim has exactly the data i want, i just would like to to export it all. Maybe as a ndjson

Thanks

**Comment by mtmail** [comment]
Afaik nobody has written such an export tool. The `nominatim export` does CSV only and doesn't contain coordinates or other data you might need.

The photon geocoder can import from a Nominatim database. https://github.com/komoot/photon/#customized-search-data So looking at the import code could be a starting point. Photon uses ElasticSearch database. Maybe it's even easier to export the ElasticSearch index into JSON, at least there seem to be tutorial how to do that in bulk.

**Reply by farfromrefug** [reply]
@mtmail thanks for the details. All that should enough for me to have a crack at it ;)
